policy-groups/infrastructure-best-practices/main.tf

```diff
@@ -1,22 +1,13 @@
-locals {
-  require_imdsv2_statement               = var.require_imdsv2 ? [""] : []
-  deny_imds_change_statement             = var.deny_imds_change ? [""] : []
-  require_ebs_encryption_statement       = var.require_ebs_encryption ? [""] : []
-  require_s3_object_encryption_statement = var.require_s3_object_encryption ? [""] : []
-  require_s3_bucket_https_statement      = var.require_s3_bucket_https ? [""] : []
-  deny_s3_public_access_statement        = var.deny_s3_public_access ? [""] : []
-  require_rds_encryption_statement       = var.require_rds_encryption ? [""] : []
-}
-
 data "aws_iam_policy_document" "combined_policy_block" {
-
   dynamic "statement" {
-    for_each = local.require_imdsv2_statement
+    for_each = var.require_imdsv2 ? [1] : []
```

Comment:
This is fine for now but it's wild that this is still how you do conditionals in modern Terraform